src/mds/journal.cc

```diff
@@ -1618,13 +1616,18 @@ void EMetaBlob::replay(MDSRank *mds, LogSegment *logseg, MDPeerUpdate *peerup)
 	if (!preallocated_inos.empty())
 	  mds->sessionmap.replay_advance_version();
       }
+
+      // repair sessionmap updates in case sessionmap wasn't persist in time
+      if (sessionmapv > mds->sessionmap.get_version()) {
+        mds->clog->error() << "EMetaBlob.replay sessionmapv mismatch "
+            << sessionmapv << " -> " << mds->sessionmap.get_version();
```

Comment:
maybe a better stmt: `"EMetaBlob.replay sessionmap v " << sessionmapv << " > " << mds->sessionmap.get_version();`

Reply:
I just copied it and followed from `inotablev` case above.

Reply:
> @lxbsz I would also like to suggest this commit message as i feel the current one needs some corrections:
> 
> ```
> When expiring the MDLog Segments it will persist the sessionmap,
> but in case the number of Segments doesn't reach up to the max
> segment threshold then the expiry may not happen.
> 
> So if there is a MDS failover then the sessionmap won't be persistent and
> cannot be replayed by the standby MDS when it is starting. We should
> just force replay the sessionmap version just the way inotable does.
> ```

Looks better and will fix it. Thanks @dparmar18